### PR TITLE
fix(iceberg): parse response data

### PR DESCRIPTION
**Pull request by marie-j**
When going through transformResponse, angular returns raw data and expect user to handle it

### CODE REVIEW

**Comment by frenautvh** [comment]
does the fix should be on api side instead ? by returning a type json response header ?

**Comment by marie-j** [comment]
> does the fix should be on api side instead ? by returning a type json response header ?

@frenautvh By default when using `transformResponse`, angular returns raw data whatever the content-type as it expects user to handle `data` the way they want

File: src/iceberg/request-upgrader.service.js [hunk 1]
```diff
@@ -108,7 +109,7 @@ export default class IcebergRequestUpgrader {
         data,
         headers,
         status,
-      ) => ({ data, headers: headers(), status }),
+      ) => ({ data: angular.fromJson(data), headers: headers(), status }),
```
Comment (FredericEspiau) [inline src/iceberg/request-upgrader.service.js]:
```suggestion
      ) => ({ data: JSON.parse(data), headers: headers(), status }),
```
?

Comment (marie-j) [reply src/iceberg/request-upgrader.service.js]: What if `data` is not parsable JSON ?

Comment (jleveugle) [reply src/iceberg/request-upgrader.service.js]: @marie-j @FredericEspiau 

It's the default behaviour when transformResponse is not override:
https://docs.angularjs.org/api/ngResource/service/$resource

So let use `angular.fromJson` ;)

**Pull request merged**